Add CRUD for Content

twitu opened the pull request:
Deleting Content also deletes all related Steps.

twitu committed:
Add CRUD for Content

dusnat commented:
There should not be an option to delete content. Since content can be shared across different pathways (not all of which have the same owner), deleting the content node would affect all those pathways unexpectedly.

Here's what I suggest as an alternative - 

Delete only steps. Whenever deleting a step, get the content node from that step and see how many edges that node has. If it has zero edges, only then delete the content node (since then we can be sure that nobody else is using the same content node).

twitu committed:
Remove delete content query

twitu committed:
Add title field to content

twitu committed:
Update Pathway and Content API

Change API to make Id from client side mandatory
Make fork node API
create update pathway uses apoc

twitu commented:
https://github.com/dush-t/pathways/wiki/Seeding-the-database
Wiki page for seeding database using updated queries.

hedonhermdev commented on server/src/db/queries/pathway/createUpdatePathway.js:
```diff
@@ -2,54 +2,53 @@ const Query = require('../../core/Query')
 
 const createUpdatePathway = ({id, name, steps}) => {
-    // create id for path if doesn't exist
-    if (!id) {
```
This is unnecessary is ID is a mandatory field right?

twitu replied:
Yeah id is mandatory. That's why I've removed it. The query will not reach here without and id value.

hedonhermdev replied:
Sorry, my bad

hedonhermdev commented:
In what situation are we gonna call `deleteContent`?
The users will not be able to call it right?

twitu commented:
Delete content is a cleanup/maintenance operation, to remove unrelated content that has not been used for a long time. It can be manual or automated. There are two ways to go about it,

1. A private API implemented using directives that only allow admins to use it.
2. Directly run cypher queries from the db interface

Either way it's not the most relevant now. Can be added later.
